### DISCUSSION TITLE
Tailwindcss CLI - purge - use only html & JS part of Vue files

### DISCUSSION THREAD

**Discussion by LouisMazel** [tailwindlabs/tailwindcss#10044] [Help] [answered]
Hey ! 

In my libraries of components Vue.JS, to get the css code I use Tailwind CLI with this cmd:

```bash
tailwindcss -i path/to/tailwind.css -o output/path/main.css --config tailwind.config.js --postcss --minify
```

But this cmd read the whole files and add in the output file all classes found in HTML templates, scripts and styles part.

An example of a Vue component with classes in each part:

```vue
<template>
  <p class="m-spinner absolute" :class="tailwindcssClass">
    Tailwind Example
  </p>
</template>

<script lang="ts" setup>
  const tailwindcssClass = 'pb-4'
</script>

<style lang="postcss" scoped>
  .m-spinner {
    @apply animate-spin fill-current text-white text-primary;
  }
</style>
```

I don't need any classes present in the CSS part because my bundler (Rollup with postcss) does the job. So, the css is already included in the component when it is imported.
I just want to get CSS in the HTML and JS part.

**So, it's possible to filter part of files to minimize the css main file ?**

**Comment by wongjn** [comment]
You could try using the [`content.transform` configuration](https://tailwindcss.com/docs/content-configuration#transforming-source-files) to modify the source code that Tailwind reads. For your situation, you could use a regular expression to remove a substring between `<style>` tags. Do bear in mind that the `transform` is based on file extension so it would affect all `.vue` files.

**Comment by LouisMazel** [comment, marked as answer]
Thanks a lot @wongjn ! It's the good way ! 

For people who's want the solution, you can exclude by replacing the content between `<style> </style>` with a regex :

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  mode: 'build',
  presets: [
    ....
  ],
  content: {
    files: [
      './your_package/**/*',
    ],
    transform: {
      vue: (content) => {
        const regex = /<style[^>]*>([\S\s]*?)<\/style>/g
        return content.replace(regex, '')
      },
    },
  },
}
```

**Comment by padcom** [comment]
If you switch to method syntax and not field initializer with lambda, you'll also get proper typing with TypeScript (in VSCode too)
